### PR TITLE
opal_show_help() cleanups.

### PR DESCRIPTION
**Pull request by awlauria**
1. Ensure it can print something when not initialized.
   While it in theory shouldn't be called before opal_show_help_init(),
   it is very easy to do so. In the case it isn't, just print to stderr.

2. Move some common code to static helper functions,
   and fix a bug where the "Sorry!" message will print '(null)' for the filename.

Signed-off-by: Austen Lauria <awlauria@us.ibm.com>

### CODE REVIEW

File: opal/util/show_help.c [hunk 1]
```diff
@@ -229,46 +248,54 @@ static int open_file(const char *base, const char *topic)
...
             filename = opal_os_path(false, search_dirs[i], base, NULL);
-            opal_show_help_yyin = fopen(filename, "r");
-            if (NULL == opal_show_help_yyin) {
-                opal_asprintf(&err_msg, "%s: %s", filename, strerror(errno));
-                base_len = strlen(base);
-                if (4 > base_len || 0 != strcmp(base + base_len - 4, ".txt")) {
-                    free(filename);
-                    opal_asprintf(&filename, "%s%s%s.txt", search_dirs[i], OPAL_PATH_SEP, base);
-                    opal_show_help_yyin = fopen(filename, "r");
+            if(filename) {
```
Comment (jsquyres) [inline opal/util/show_help.c:251]: Is there a case where `filename==NULL` here?

Comment (awlauria) [reply opal/util/show_help.c:251]: Yes - in the case where `opal_show_help()` was called before `opal_show_help_init()` - `search_dirs` won't be populated yet.

Comment (awlauria) [reply opal/util/show_help.c:251]: Probably not - I added it for extra caution. I can remove it if desired.

Comment (jsquyres) [reply opal/util/show_help.c:251]: I guess "defensive programming" is a valid reason.

**Review by jsquyres — APPROVED** [review]

**Pull request merged**